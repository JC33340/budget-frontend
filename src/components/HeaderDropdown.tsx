import { IoIosMenu } from 'react-icons/io';
import { Link } from 'react-router-dom';
import { useState } from 'react';

const HeaderDropdown = () => {
    //hidden state
    const [hidden, setHidden] = useState(true);
    const options = [
        { name: 'Transactions', href: '/transactions' },
        { name: 'Reports', href: '/reports' },
    ];

    return (
        <div
            className="flex md:hidden relative"
            data-testid="headerDropdown-testid"
        >
            <IoIosMenu
                className="fill-white cursor-pointer"
                size={40}
                onClick={() => setHidden((prev) => !prev)}
                data-testid="dropdownButton-testid"
            />
            <div
                className={`absolute ${hidden ? 'hidden' : 'flex'} flex-col right-0 bottom-[-190px] rounded-lg border-4 overflow-hidden border-dark-gray`}
                data-testid="dropdownContent-testid"
            >
                {options.map((item, i) => {
                    return (
                        <Link
                            key={i}
                            className={`cursor-pointer border-b-2 bg-white p-2 text-lg text-blue font-semibold hover:bg-blue hover:text-white transition-colors ease-in-out`}
                            to={item.href}
                        >
                            {item.name}
                        </Link>
                    );
                })}
                <button
                    className="cursor-pointer bg-dark-red p-2 text-lg text-white font-semibold"
                    onClick={() => {
                        localStorage.clear();
                        return window.location.reload();
                    }}
                >
                    Log out
                </button>
            </div>
        </div>
    );
};

export default HeaderDropdown;
