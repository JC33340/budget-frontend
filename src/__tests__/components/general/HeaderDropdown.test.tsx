import HeaderDropdown from '../../../components/HeaderDropdown';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

test('snapshot test', () => {
    render(
        <BrowserRouter>
            <HeaderDropdown />
        </BrowserRouter>
    );
    const dropdown = screen.getByTestId('headerDropdown-testid');
    expect(dropdown).toMatchSnapshot();
});

test('testing hidden function', async () => {
    //setup user event
    const user = userEvent.setup();
    render(
        <BrowserRouter>
            <HeaderDropdown />
        </BrowserRouter>
    );
    const dropdown = screen.getByTestId('dropdownContent-testid');
    //expect attribute to be hidden
    expect(dropdown).toHaveClass('hidden');
    const dropdownButton = screen.getByTestId('dropdownButton-testid');
    await user.click(dropdownButton);
    expect(dropdown).toHaveClass('flex');
});
