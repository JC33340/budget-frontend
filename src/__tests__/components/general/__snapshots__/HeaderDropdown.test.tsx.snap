// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  class="flex md:hidden relative"
  data-testid="headerDropdown-testid"
>
  <svg
    class="fill-white cursor-pointer"
    data-testid="dropdownButton-testid"
    fill="currentColor"
    height="40"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 512 512"
    width="40"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M432 176H80c-8.8 0-16-7.2-16-16s7.2-16 16-16h352c8.8 0 16 7.2 16 16s-7.2 16-16 16zM432 272H80c-8.8 0-16-7.2-16-16s7.2-16 16-16h352c8.8 0 16 7.2 16 16s-7.2 16-16 16zM432 368H80c-8.8 0-16-7.2-16-16s7.2-16 16-16h352c8.8 0 16 7.2 16 16s-7.2 16-16 16z"
    />
  </svg>
  <div
    class="absolute hidden flex-col right-0 bottom-[-190px] rounded-lg border-4 overflow-hidden border-dark-gray"
    data-testid="dropdownContent-testid"
  >
    <a
      class="cursor-pointer border-b-2 bg-white p-2 text-lg text-blue font-semibold hover:bg-blue hover:text-white transition-colors ease-in-out"
      href="/transactions"
    >
      Transactions
    </a>
    <a
      class="cursor-pointer border-b-2 bg-white p-2 text-lg text-blue font-semibold hover:bg-blue hover:text-white transition-colors ease-in-out"
      href="/reports"
    >
      Reports
    </a>
    <a
      class="cursor-pointer border-b-2 bg-white p-2 text-lg text-blue font-semibold hover:bg-blue hover:text-white transition-colors ease-in-out"
      href="/settings"
    >
      Settings
    </a>
    <button
      class="cursor-pointer bg-dark-red p-2 text-lg text-white font-semibold"
    >
      Log out
    </button>
  </div>
</div>
`;
